packages/scene-composer/src/utils/entityModelUtils/modelRefComponent.ts

```diff
@@ -63,8 +63,35 @@ export const createModelRefEntityComponent = (model: IModelRefComponent): Compon
   return comp;
 };
 
-export const updateModelRefEntityComponent = (model: IModelRefComponent): ComponentUpdateRequest => {
+export const updateModelRefEntityComponent = (
+  model: IModelRefComponent,
+  oldComponent?: IModelRefComponent,
+): ComponentUpdateRequest => {
   const request = createModelRefEntityComponent(model);
+  if (oldComponent) {
```

Comment:
Don't need this `if (oldComponent)` check as the subsequent checks cover the logic.